src/components/PopupWithForm.js

```diff
@@ -8,26 +8,37 @@ export default class PopupWithForm extends Popup {
     this._inputs = Array.from(
       this._popupForm.querySelectorAll(".modal__input")
     );
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._submitButtonText =
+      this._submitButton.childNodes[0].textContent.trim();
   }
 
   _getInputValues() {
     this._inputValues = {};
     this._inputs.forEach((input) => {
       this._inputValues[input.name] = input.value;
     });
-
     return this._inputValues;
   }
 
   setEventListeners() {
     super.setEventListeners();
 
-    this._submitHandler = (evt) => {
+    this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
+      this.renderLoading(true);
+      this._handleFormSubmit(this._getInputValues())
+        .then(() => this.close())
+        .catch((err) => console.error("Error submitting form:", err))
+        .finally(() => this.renderLoading(false));
+    });
+  }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`